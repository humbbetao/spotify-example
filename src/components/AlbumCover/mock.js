import { mount } from 'enzyme'
import React from 'react'
import { createMockStore } from 'redux-test-utils'
import { Provider } from 'react-redux'
import { Router, BrowserRouter } from 'react-router-dom'
import { createHistory } from 'react-router-dom'
import PropTypes from 'prop-types'

export const mountWithRouterAndStore = (Component, storeState = {}) => {
  const store = createMockStore(storeState)

  return mount(
    <Provider store={store}>
      <BrowserRouter>{Component}</BrowserRouter>
    </Provider>
  )
}

export function mountWithRouterAndStore2(Component, storeState = {}) {
  const store = createMockStore(storeState)
  // const history = createHistory()
  const history = {}
  const options = {
    context: { store, router: { history, route: { location: {} } } },
    childContextTypes: {
      store: PropTypes.object.isRequired,
      router: PropTypes.object.isRequired,
    },
  }

  return mount(Component, options)
}
