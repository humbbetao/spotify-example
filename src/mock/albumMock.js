const arrayMock = [
  {
    album_type: 'single',
    artists: [
      {
        external_urls: {
          spotify: 'https://open.spotify.com/artist/6rFkZxhheU9l1nODgZm4SP',
        },
        href: 'https://api.spotify.com/v1/artists/6rFkZxhheU9l1nODgZm4SP',
        id: '6rFkZxhheU9l1nODgZm4SP',
        name: 'Junior Lord',
        type: 'artist',
        uri: 'spotify:artist:6rFkZxhheU9l1nODgZm4SP',
      },
    ],
    available_markets: [
      'AD',
      'AE',
      'AR',
      'AT',
      'AU',
      'BE',
      'BG',
      'BH',
      'BO',
      'BR',
      'CA',
      'CH',
      'CL',
      'CO',
      'CR',
      'CY',
      'CZ',
      'DE',
      'DK',
      'DO',
      'DZ',
      'EC',
      'EE',
      'EG',
      'ES',
      'FI',
      'FR',
      'GB',
      'GR',
      'GT',
      'HK',
      'HN',
      'HU',
      'ID',
      'IE',
      'IL',
      'IN',
      'IS',
      'IT',
      'JO',
      'JP',
      'KW',
      'LB',
      'LI',
      'LT',
      'LU',
      'LV',
      'MA',
      'MC',
      'MT',
      'MX',
      'MY',
      'NI',
      'NL',
      'NO',
      'NZ',
      'OM',
      'PA',
      'PE',
      'PH',
      'PL',
      'PS',
      'PT',
      'PY',
      'QA',
      'RO',
      'SA',
      'SE',
      'SG',
      'SK',
      'SV',
      'TH',
      'TN',
      'TR',
      'TW',
      'US',
      'UY',
      'VN',
      'ZA',
    ],
    external_urls: {
      spotify: 'https://open.spotify.com/album/2Yv3YuHlHmeSnMYdF5hhzS',
    },
    href: 'https://api.spotify.com/v1/albums/2Yv3YuHlHmeSnMYdF5hhzS',
    id: '2Yv3YuHlHmeSnMYdF5hhzS',
    images: [
      {
        height: 640,
        url: 'https://i.scdn.co/image/ab67616d0000b273c2744f6a4677f1677b3eca12',
        width: 640,
      },
      {
        height: 300,
        url: 'https://i.scdn.co/image/ab67616d00001e02c2744f6a4677f1677b3eca12',
        width: 300,
      },
      {
        height: 64,
        url: 'https://i.scdn.co/image/ab67616d00004851c2744f6a4677f1677b3eca12',
        width: 64,
      },
    ],
    name: 'Rihanna',
    release_date: '2019-06-21',
    release_date_precision: 'day',
    total_tracks: 1,
    type: 'album',
    uri: 'spotify:album:2Yv3YuHlHmeSnMYdF5hhzS',
  },
  {
    album_type: 'album',
    artists: [
      {
        external_urls: {
          spotify: 'https://open.spotify.com/artist/5pKCCKE2ajJHZ9KAiaK11H',
        },
        href: 'https://api.spotify.com/v1/artists/5pKCCKE2ajJHZ9KAiaK11H',
        id: '5pKCCKE2ajJHZ9KAiaK11H',
        name: 'Rihanna',
        type: 'artist',
        uri: 'spotify:artist:5pKCCKE2ajJHZ9KAiaK11H',
      },
    ],
    available_markets: [
      'AD',
      'AE',
      'AR',
      'AT',
      'AU',
      'BE',
      'BG',
      'BH',
      'BO',
      'BR',
      'CA',
      'CH',
      'CL',
      'CO',
      'CR',
      'CY',
      'CZ',
      'DE',
      'DK',
      'DO',
      'DZ',
      'EC',
      'EE',
      'EG',
      'ES',
      'FI',
      'FR',
      'GB',
      'GR',
      'GT',
      'HK',
      'HN',
      'HU',
      'ID',
      'IE',
      'IL',
      'IN',
      'IS',
      'IT',
      'JO',
      'JP',
      'KW',
      'LB',
      'LI',
      'LT',
      'LU',
      'LV',
      'MA',
      'MC',
      'MT',
      'MX',
      'MY',
      'NI',
      'NL',
      'NO',
      'NZ',
      'OM',
      'PA',
      'PE',
      'PH',
      'PL',
      'PS',
      'PT',
      'PY',
      'QA',
      'RO',
      'SA',
      'SE',
      'SG',
      'SK',
      'SV',
      'TH',
      'TN',
      'TR',
      'TW',
      'US',
      'UY',
      'VN',
      'ZA',
    ],
    external_urls: {
      spotify: 'https://open.spotify.com/album/4UlGauD7ROb3YbVOFMgW5u',
    },
    href: 'https://api.spotify.com/v1/albums/4UlGauD7ROb3YbVOFMgW5u',
    id: '4UlGauD7ROb3YbVOFMgW5u',
    images: [
      {
        height: 640,
        url: 'https://i.scdn.co/image/ab67616d0000b27333c6b920eabcf4c00d7a1093',
        width: 640,
      },
      {
        height: 300,
        url: 'https://i.scdn.co/image/ab67616d00001e0233c6b920eabcf4c00d7a1093',
        width: 300,
      },
      {
        height: 64,
        url: 'https://i.scdn.co/image/ab67616d0000485133c6b920eabcf4c00d7a1093',
        width: 64,
      },
    ],
    name: 'ANTI (Deluxe)',
    release_date: '2016-01-28',
    release_date_precision: 'day',
    total_tracks: 16,
    type: 'album',
    uri: 'spotify:album:4UlGauD7ROb3YbVOFMgW5u',
  },
  {
    album_type: 'album',
    artists: [
      {
        external_urls: {
          spotify: 'https://open.spotify.com/artist/5pKCCKE2ajJHZ9KAiaK11H',
        },
        href: 'https://api.spotify.com/v1/artists/5pKCCKE2ajJHZ9KAiaK11H',
        id: '5pKCCKE2ajJHZ9KAiaK11H',
        name: 'Rihanna',
        type: 'artist',
        uri: 'spotify:artist:5pKCCKE2ajJHZ9KAiaK11H',
      },
    ],
    available_markets: [
      'AD',
      'AE',
      'AR',
      'AT',
      'AU',
      'BE',
      'BG',
      'BH',
      'BO',
      'BR',
      'CA',
      'CH',
      'CL',
      'CO',
      'CR',
      'CY',
      'CZ',
      'DE',
      'DK',
      'DO',
      'DZ',
      'EC',
      'EE',
      'EG',
      'ES',
      'FI',
      'FR',
      'GB',
      'GR',
      'GT',
      'HK',
      'HN',
      'HU',
      'ID',
      'IE',
      'IL',
      'IN',
      'IS',
      'IT',
      'JO',
      'JP',
      'KW',
      'LB',
      'LI',
      'LT',
      'LU',
      'LV',
      'MA',
      'MC',
      'MT',
      'MX',
      'MY',
      'NI',
      'NL',
      'NO',
      'NZ',
      'OM',
      'PA',
      'PE',
      'PH',
      'PL',
      'PS',
      'PT',
      'PY',
      'QA',
      'RO',
      'SA',
      'SE',
      'SG',
      'SK',
      'SV',
      'TH',
      'TN',
      'TR',
      'TW',
      'US',
      'UY',
      'VN',
      'ZA',
    ],
    external_urls: {
      spotify: 'https://open.spotify.com/album/3JSWZWeTHF4HDGt5Eozdy7',
    },
    href: 'https://api.spotify.com/v1/albums/3JSWZWeTHF4HDGt5Eozdy7',
    id: '3JSWZWeTHF4HDGt5Eozdy7',
    images: [
      {
        height: 640,
        url: 'https://i.scdn.co/image/ab67616d0000b273f9f27162ab1ed45b8d7a7e98',
        width: 640,
      },
      {
        height: 300,
        url: 'https://i.scdn.co/image/ab67616d00001e02f9f27162ab1ed45b8d7a7e98',
        width: 300,
      },
      {
        height: 64,
        url: 'https://i.scdn.co/image/ab67616d00004851f9f27162ab1ed45b8d7a7e98',
        width: 64,
      },
    ],
    name: 'Good Girl Gone Bad: Reloaded',
    release_date: '2008-06-02',
    release_date_precision: 'day',
    total_tracks: 15,
    type: 'album',
    uri: 'spotify:album:3JSWZWeTHF4HDGt5Eozdy7',
  },
  {
    album_type: 'album',
    artists: [
      {
        external_urls: {
          spotify: 'https://open.spotify.com/artist/5pKCCKE2ajJHZ9KAiaK11H',
        },
        href: 'https://api.spotify.com/v1/artists/5pKCCKE2ajJHZ9KAiaK11H',
        id: '5pKCCKE2ajJHZ9KAiaK11H',
        name: 'Rihanna',
        type: 'artist',
        uri: 'spotify:artist:5pKCCKE2ajJHZ9KAiaK11H',
      },
    ],
    available_markets: [
      'AD',
      'AE',
      'AR',
      'AT',
      'AU',
      'BE',
      'BG',
      'BH',
      'BO',
      'BR',
      'CH',
      'CL',
      'CO',
      'CR',
      'CY',
      'CZ',
      'DE',
      'DK',
      'DO',
      'DZ',
      'EC',
      'EE',
      'EG',
      'ES',
      'FI',
      'FR',
      'GB',
      'GR',
      'GT',
      'HK',
      'HN',
      'HU',
      'ID',
      'IL',
      'IN',
      'IS',
      'IT',
      'JO',
      'JP',
      'KW',
      'LB',
      'LI',
      'LT',
      'LU',
      'LV',
      'MA',
      'MC',
      'MT',
      'MY',
      'NI',
      'NL',
      'NO',
      'NZ',
      'OM',
      'PA',
      'PE',
      'PH',
      'PL',
      'PS',
      'PT',
      'PY',
      'QA',
      'RO',
      'SA',
      'SE',
      'SG',
      'SK',
      'SV',
      'TH',
      'TN',
      'TR',
      'TW',
      'UY',
      'VN',
      'ZA',
    ],
    external_urls: {
      spotify: 'https://open.spotify.com/album/7vN82vd1Vq44fjlhjfvHJp',
    },
    href: 'https://api.spotify.com/v1/albums/7vN82vd1Vq44fjlhjfvHJp',
    id: '7vN82vd1Vq44fjlhjfvHJp',
    images: [
      {
        height: 640,
        url: 'https://i.scdn.co/image/ab67616d0000b273be99e0ef9d3e6a1b55d94e78',
        width: 640,
      },
      {
        height: 300,
        url: 'https://i.scdn.co/image/ab67616d00001e02be99e0ef9d3e6a1b55d94e78',
        width: 300,
      },
      {
        height: 64,
        url: 'https://i.scdn.co/image/ab67616d00004851be99e0ef9d3e6a1b55d94e78',
        width: 64,
      },
    ],
    name: 'Loud',
    release_date: '2010-11-16',
    release_date_precision: 'day',
    total_tracks: 12,
    type: 'album',
    uri: 'spotify:album:7vN82vd1Vq44fjlhjfvHJp',
  },
]

export default arrayMock
