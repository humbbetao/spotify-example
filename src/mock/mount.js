import { mount } from 'enzyme'
import React from 'react'
import { createMockStore } from 'redux-test-utils'
import { Provider } from 'react-redux'
import { Router, BrowserRouter, Switch } from 'react-router-dom'
import PropTypes from 'prop-types'

export const mountWithRouterAndStore = (Component, storeState = {}) => {
  const store = createMockStore(storeState)

  return mount(
    <Provider store={store}>
      <BrowserRouter>{Component}</BrowserRouter>
    </Provider>
  )
}

export const mountWithRouterAndStoreWithRoutes = (
  Component,
  storeState = {},
  path
) => {
  const store = createMockStore(storeState)

  return mount(
    <Provider store={store}>
      <BrowserRouter>
        <Router path={path} Component={Component}></Router>
      </BrowserRouter>
    </Provider>
  )
}

export function mountWithRouterAndStore2(Component, storeState = {}) {
  const store = createMockStore(storeState)
  const history = {}
  const options = {
    context: { store, router: { history, route: { location: {} } } },
    childContextTypes: {
      store: PropTypes.object.isRequired,
      router: PropTypes.object.isRequired,
    },
  }

  return mount(Component, options)
}
