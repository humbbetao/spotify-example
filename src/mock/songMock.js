const arrayMock = [
  {
    artists: [
      {
        external_urls: {
          spotify: 'https://open.spotify.com/artist/5pKCCKE2ajJHZ9KAiaK11H',
        },
        href: 'https://api.spotify.com/v1/artists/5pKCCKE2ajJHZ9KAiaK11H',
        id: '5pKCCKE2ajJHZ9KAiaK11H',
        name: 'Rihanna',
        type: 'artist',
        uri: 'spotify:artist:5pKCCKE2ajJHZ9KAiaK11H',
      },
      {
        external_urls: {
          spotify: 'https://open.spotify.com/artist/7tYKF4w9nC0nq9CsPZTHyP',
        },
        href: 'https://api.spotify.com/v1/artists/7tYKF4w9nC0nq9CsPZTHyP',
        id: '7tYKF4w9nC0nq9CsPZTHyP',
        name: 'SZA',
        type: 'artist',
        uri: 'spotify:artist:7tYKF4w9nC0nq9CsPZTHyP',
      },
    ],
    available_markets: [
      'AD',
      'AE',
      'AR',
      'AT',
      'AU',
      'BE',
      'BG',
      'BH',
      'BO',
      'BR',
      'CA',
      'CH',
      'CL',
      'CO',
      'CR',
      'CY',
      'CZ',
      'DE',
      'DK',
      'DO',
      'DZ',
      'EC',
      'EE',
      'EG',
      'ES',
      'FI',
      'FR',
      'GB',
      'GR',
      'GT',
      'HK',
      'HN',
      'HU',
      'ID',
      'IE',
      'IL',
      'IN',
      'IS',
      'IT',
      'JO',
      'JP',
      'KW',
      'LB',
      'LI',
      'LT',
      'LU',
      'LV',
      'MA',
      'MC',
      'MT',
      'MX',
      'MY',
      'NI',
      'NL',
      'NO',
      'NZ',
      'OM',
      'PA',
      'PE',
      'PH',
      'PL',
      'PS',
      'PT',
      'PY',
      'QA',
      'RO',
      'SA',
      'SE',
      'SG',
      'SK',
      'SV',
      'TH',
      'TN',
      'TR',
      'TW',
      'US',
      'UY',
      'VN',
      'ZA',
    ],
    disc_number: 1,
    duration_ms: 161066,
    explicit: true,
    external_urls: {
      spotify: 'https://open.spotify.com/track/6t90Z9XkdsHD8xMxro6KRP',
    },
    href: 'https://api.spotify.com/v1/tracks/6t90Z9XkdsHD8xMxro6KRP',
    id: '6t90Z9XkdsHD8xMxro6KRP',
    is_local: false,
    name: 'Consideration',
    preview_url:
      'https://p.scdn.co/mp3-preview/5ed5f5f05f603c7ef1ad5f03abf6d7b8eeafd03e?cid=95555d82a874400ab026efab926d00ae',
    track_number: 1,
    type: 'track',
    uri: 'spotify:track:6t90Z9XkdsHD8xMxro6KRP',
  },
  {
    artists: [
      {
        external_urls: {
          spotify: 'https://open.spotify.com/artist/5pKCCKE2ajJHZ9KAiaK11H',
        },
        href: 'https://api.spotify.com/v1/artists/5pKCCKE2ajJHZ9KAiaK11H',
        id: '5pKCCKE2ajJHZ9KAiaK11H',
        name: 'Rihanna',
        type: 'artist',
        uri: 'spotify:artist:5pKCCKE2ajJHZ9KAiaK11H',
      },
    ],
    available_markets: [
      'AD',
      'AE',
      'AR',
      'AT',
      'AU',
      'BE',
      'BG',
      'BH',
      'BO',
      'BR',
      'CA',
      'CH',
      'CL',
      'CO',
      'CR',
      'CY',
      'CZ',
      'DE',
      'DK',
      'DO',
      'DZ',
      'EC',
      'EE',
      'EG',
      'ES',
      'FI',
      'FR',
      'GB',
      'GR',
      'GT',
      'HK',
      'HN',
      'HU',
      'ID',
      'IE',
      'IL',
      'IN',
      'IS',
      'IT',
      'JO',
      'JP',
      'KW',
      'LB',
      'LI',
      'LT',
      'LU',
      'LV',
      'MA',
      'MC',
      'MT',
      'MX',
      'MY',
      'NI',
      'NL',
      'NO',
      'NZ',
      'OM',
      'PA',
      'PE',
      'PH',
      'PL',
      'PS',
      'PT',
      'PY',
      'QA',
      'RO',
      'SA',
      'SE',
      'SG',
      'SK',
      'SV',
      'TH',
      'TN',
      'TR',
      'TW',
      'US',
      'UY',
      'VN',
      'ZA',
    ],
    disc_number: 1,
    duration_ms: 72080,
    explicit: true,
    external_urls: {
      spotify: 'https://open.spotify.com/track/6K4Co1NQyOnoFEJMkZHkln',
    },
    href: 'https://api.spotify.com/v1/tracks/6K4Co1NQyOnoFEJMkZHkln',
    id: '6K4Co1NQyOnoFEJMkZHkln',
    is_local: false,
    name: 'James Joint',
    preview_url:
      'https://p.scdn.co/mp3-preview/8fc2a6771f56a60683979bd876bc6b73f25052a1?cid=95555d82a874400ab026efab926d00ae',
    track_number: 2,
    type: 'track',
    uri: 'spotify:track:6K4Co1NQyOnoFEJMkZHkln',
  },
  {
    artists: [
      {
        external_urls: {
          spotify: 'https://open.spotify.com/artist/5pKCCKE2ajJHZ9KAiaK11H',
        },
        href: 'https://api.spotify.com/v1/artists/5pKCCKE2ajJHZ9KAiaK11H',
        id: '5pKCCKE2ajJHZ9KAiaK11H',
        name: 'Rihanna',
        type: 'artist',
        uri: 'spotify:artist:5pKCCKE2ajJHZ9KAiaK11H',
      },
    ],
    available_markets: [
      'AD',
      'AE',
      'AR',
      'AT',
      'AU',
      'BE',
      'BG',
      'BH',
      'BO',
      'BR',
      'CA',
      'CH',
      'CL',
      'CO',
      'CR',
      'CY',
      'CZ',
      'DE',
      'DK',
      'DO',
      'DZ',
      'EC',
      'EE',
      'EG',
      'ES',
      'FI',
      'FR',
      'GB',
      'GR',
      'GT',
      'HK',
      'HN',
      'HU',
      'ID',
      'IE',
      'IL',
      'IN',
      'IS',
      'IT',
      'JO',
      'JP',
      'KW',
      'LB',
      'LI',
      'LT',
      'LU',
      'LV',
      'MA',
      'MC',
      'MT',
      'MX',
      'MY',
      'NI',
      'NL',
      'NO',
      'NZ',
      'OM',
      'PA',
      'PE',
      'PH',
      'PL',
      'PS',
      'PT',
      'PY',
      'QA',
      'RO',
      'SA',
      'SE',
      'SG',
      'SK',
      'SV',
      'TH',
      'TN',
      'TR',
      'TW',
      'US',
      'UY',
      'VN',
      'ZA',
    ],
    disc_number: 1,
    duration_ms: 253080,
    explicit: true,
    external_urls: {
      spotify: 'https://open.spotify.com/track/0UHB9METy4VCXNgkcGqHqS',
    },
    href: 'https://api.spotify.com/v1/tracks/0UHB9METy4VCXNgkcGqHqS',
    id: '0UHB9METy4VCXNgkcGqHqS',
    is_local: false,
    name: 'Kiss It Better',
    preview_url:
      'https://p.scdn.co/mp3-preview/5857fe6cdb7597f6da72c3951696086c042045ed?cid=95555d82a874400ab026efab926d00ae',
    track_number: 3,
    type: 'track',
    uri: 'spotify:track:0UHB9METy4VCXNgkcGqHqS',
  },
]
export default arrayMock
