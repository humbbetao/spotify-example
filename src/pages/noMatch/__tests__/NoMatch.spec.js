import React from 'react'
import NoMatch from '../NoMatch'
import { mountWithRouterAndStore } from 'mock/mount'

describe('NoMatch', () => {
  it('should render correctly', () => {
    const wrapper = mountWithRouterAndStore(<NoMatch />)

    expect(wrapper).toMatchSnapshot()
  })
})
