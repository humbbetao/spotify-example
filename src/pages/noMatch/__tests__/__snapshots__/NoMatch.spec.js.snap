// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoMatch should render correctly 1`] = `ReactWrapper {}`;
